src/dune_rules/melange/melange_rules.ml

```diff
@@ -502,6 +509,13 @@ let setup_js_rules_libraries =
       let* includes =
         let+ requires_link =
           Memo.Lazy.force (Lib.Compile.requires_link lib_compile_info)
+          |> Resolve.Memo.map ~f:(fun deps ->
```

Comment:
iiuc using `map` will effectively replace the vlib dep with its implementation, for the sake of the cmj deps. But what would happen if the vlib itself contains some `ml` files?

Edit: never mind, I see below that we start from the existing `deps` and the addition of the implementations is purely additive.